### PR TITLE
Feature/site detail page

### PR DESCRIPTION
**Pull request by ericfennis**
## Icon Detail page!

- When you click now, your route will change and opens up the icon.
- Before it downloaded the icon, now you can choose if you want to Donwload or Copy.
- The key shortcuts still work when you click on an icon. CMD/ctrl + click and shift + click (maybe we need an hint for this somewhere on the website)
- I refactored and split up some code.
- @aelfric Your code works awesome, maybe some code of you is moved in to components.

localhost_3000_icon_folder-plus

**DEMO:**[deploy-preview-99--lucide.netlify.app]( https://deploy-preview-99--lucide.netlify.app/)

### CODE REVIEW

**Comment by locness3** [comment]
The website is missing the "context".

**Comment by aelfric** [comment]
Is there something specific you want me to look at?

**Comment by ericfennis** [comment]
@aelfric no sorry, Just want to inform you.

**Comment by ericfennis** [comment]
@locness3 text is added again!

File: site/src/components/Header.tsx [hunk 1]
```diff
@@ -25,8 +26,11 @@ const Header = ({ data }) => {
 
   return (
     <Flex direction="column" align="center" justify="center">
-      <Text fontSize="3xl" as="b" textAlign="center">
-        Simply beautiful open source icons, community-sourced.
+      <Text fontSize="3xl" as="b" mb="4" textAlign="center">
+        Simply beautiful open source icons, community-sourced
+      </Text>
+      <Text fontSize="lg" as="p" textAlign="center" mb="8">
+        An open-source icon library, a fork of Feather Icons. <br/>We're expanding the icon set as much as possible while keeping it nice-looking - <Link href="https://github.com/lucide-icons/lucide" isExternal>join us</Link>!
```
Comment (locness3) [inline site/src/components/Header.tsx:33]: `Feather Icons` should be a link as well

**Comment by delnyn** [comment]
The search bar is on the top of the details 
image

**Comment by delnyn** [comment]
Something is wrong at `/src/components/Layout.tsx:52:66`... "`Type error: No overload matches this call.`"

**Comment by ericfennis** [comment]
Working on it!

**Comment by ericfennis** [comment]
I Updated the site.
I also fixed the issue for #101 by updating it to a pre-release of Chakra UI where it is fixed, they used as wel on their own website.
One thing that I needed to change was the Icons in buttons, I used a beta release of our package. Which work fine I noticed. Later on I will update update the rest of the icons if I'm done with building it.

**Comment by delnyn** [comment]
Great! But I tested the preview a bit and I think it's not detecting the system theme (after deleting local storage and reloading, it's not the system theme)

**Comment by ericfennis** [comment]
Yeah you right, I forgot to test that. I will make an quick update for that tomorrow.

**Comment by ericfennis** [comment]
It's working now, chakra fixed it!

**Comment by delnyn** [comment]
Great 👏  
What is next?

**Comment by ericfennis** [comment]
Merging!

**Pull request merged**